src/pages/resultsView/pathwayMapper/ResultsViewPathwayMapper.tsx

```diff
@@ -0,0 +1,198 @@
+                              this.props.store.selectedMolecularProfiles.result!,
+                              alterationData,
+                              true,
+                              this.alterationFrequencyData,
+                              this.props.store.genes.result!);
+        });
+
+
+        // Alteration data of non-query genes are loaded.
+        const isNewStoreReady = this.storeForAllData &&
+                                this.storeForAllData.oqlFilteredCaseAggregatedDataByUnflattenedOQLLine.isComplete &&
+                                this.storeForAllData.samples.isComplete &&
+                                this.storeForAllData.patients.isComplete &&
+                                this.storeForAllData.coverageInformation.isComplete &&
+                                this.storeForAllData.sequencedSampleKeysByGene.isComplete &&
+                                this.storeForAllData.sequencedPatientKeysByGene.isComplete &&
+                                this.storeForAllData.selectedMolecularProfiles.isComplete;
+
+        if(isNewStoreReady){
+
+            this.storeForAllData!.oqlFilteredCaseAggregatedDataByUnflattenedOQLLine.result!.forEach( (alterationData, trackIndex) => {
+                getAlterationData(this.storeForAllData!.samples.result,
+                                  this.storeForAllData!.patients.result,
+                                  this.storeForAllData!.coverageInformation.result,
+                                  this.storeForAllData!.sequencedSampleKeysByGene.result!,
+                                  this.storeForAllData!.sequencedPatientKeysByGene.result!,
+                                  this.storeForAllData!.selectedMolecularProfiles.result!,
+                                  alterationData,
+                                  false,
+                                  this.alterationFrequencyData,
+                                  this.props.store.genes.result!);
+                
+            });
+
+            this.addGenomicData(this.alterationFrequencyData);
+            // Toasts are removed with delay
+            setTimeout(() => {this.activeToasts.forEach(tId => {toast.dismiss(tId);});}, 1000);
+        }
+
+        return(
+
+            <div className="pathwayMapper">
+            <div data-test="pathwayMapperTabDiv" className="cBioMode" style={{width: "99%"}}>
+                <Row>
+
+                    { !this.isLoading ? 
+
+                        <React.Fragment>
+                            <OqlStatusBanner className="coexp-oql-status-banner" store={this.props.store} tabReflectsOql={true}/>
+
+                            <PathwayMapper
+                                isCBioPortal={true}
+                                isCollaborative={false}
+                                genes={this.props.store.genes.result as any}
+                                cBioAlterationData={this.alterationFrequencyData}
```

Comment:
Are the props of PathwayMapper well typed?   There is dependency here that would be easy to break if not.

Reply:
Do you mean the `genes` prop? If so, since `Gene` type is not available from within PathwayMapper library, I specified it as `any` and PathwayMapper library also expects `any` type for that property.